packages/backend/src/rpc/dispatcher.spec.ts

```diff
@@ -313,6 +337,94 @@ describe('RPCDispatcher', () => {
           );
         })
         .then(() => {
+          expect(dummyHandler).not.toHaveBeenCalled();
+          expect(specificHandler).toHaveBeenCalledTimes(1);
+          expect(app.getSingleton(ServiceContextEvents).disposeContext).toHaveBeenCalled();
+        });
+    });
+
+    it('should respond with error response from the last registered error handler when the RPC call fails with error that the handler can handle', () => {
+      let app = new App();
+
+      // prepare request / response
+      app.overrideService(
+        RequestInfo,
+        class MockRequestInfo extends RequestInfo {
+          req = mockRequest('test.method', {});
+          res = mockResponse();
+        },
+      );
+
+      // mock middleware call and prepare erroneous response
+      app.overrideSingleton(
+        RPCMiddlewareContainer,
+        class MockRPCMiddlewareContainer extends RPCMiddlewareContainer {
+          call = jest.fn(() => Promise.reject({ isJoi: true, details: 'Error details' }));
+        },
+      );
+
+      // register handler that can handle the error but cannot be reached? reached
+      let specificHandler1 = jest.fn((e: Error) => {
+        if ((e as any).isJoi) {
+          return {
+            code: 400,
+            message: 'Technical error, the request was malformed.',
+            data: (e as any).details,
+          };
+        }
+      });
+
+      // register third handler with same condition as the previous to ensure it was not reached
+      let specificHandler2 = jest.fn((e: Error) => {
+        if ((e as any).isJoi) {
+          return {
+            code: 444,
+            message: 'I am the most recently registered handler and I am handling the error.',
+            data: (e as any).details,
+          };
+        }
+      });
+
+      // register the handles in order
```

Comment:
The same minor comment remarks are valid for this test as well.

Reply:
will do